sql/catalyst/src/main/scala/org/apache/spark/sql/types/Metadata.scala

```diff
@@ -273,6 +273,9 @@ class MetadataBuilder {
   /** Puts a [[Metadata]] array. */
   def putMetadataArray(key: String, value: Array[Metadata]): this.type = put(key, value)
 
+  /** Puts a name. */
+  def putName(name: String): this.type = put("name", name)
```

Comment:
This interface change is not desired. See the PR https://github.com/apache/spark/pull/16209

You can further enhance our parser by supporting the data types that are not natively supported by Spark.